### DISCUSSION TITLE
Allow user to manage their own profile

### DISCUSSION THREAD

**Discussion by aliefZafran** [zitadel/zitadel#6888] [Q&A]
Hi I am new to this open source community and also still learning on this IAM tool, hence I hope i can get some help. I am currently trying to have a feature in my react app in which would allow my users to manage their own profile without having them to access the admin console. I believe it could be done so by following this [link](https://zitadel.com/docs/apis/resources/auth/auth-service-add-my-auth-factor-otp). But upon sending the request, im actually a bit confused as to which bearer token should i send. Is it the access token obtained from the service user or the token obtained upon my user logged in to the app?

**Comment by hifabienne** [comment]
Hei @aliefZafran 
Yes the auth API is exactly for that. The users are able to manage themself with the auth API.
You do have to send the token of the user itself, and the manipulations will happen on that user.

**Reply by aliefZafran** [reply]
i see, but im a bit lost at the 'send the user's token itself' part because ive been doing it this whole time and always got return an error. Assuming if my way of obtaining the access token is correct. Now, i have a react app which i cloned from the zitadel documentation and configured it to print out the  access token upon user logged in. But upon i try to send a request to the api with the token obtained from the react app, i would always got an error. But until recently, i used a service user instead in which the access token can easily be obtained via the tutorial [here](https://zitadel.com/docs/guides/integrate/access-zitadel-apis#authenticating-a-service-user). I got the expected output but that only for my service user.... Is my way of obtaining my user's access token is correct via the react app or is there any other way of obtaining my user's access token?
